### PR TITLE
Change widget::addEventListener() to support multiple event listeners

### PR DESCRIPTION
**Pull request by enhhh**
change Widget::addEventListener() from only one event listener to multiple event listeners

We may need to add an Event Listener anytime( for example: newbie guide) and remove an Listener anytime.
so I change `_touchEventCallback` to an `std::map`, key is an `uint32_t`,return key after add an event listener, and use it to remove event listener.

### CODE REVIEW

**Comment by zilongshanren** [comment]
Thanks @enhhh 
Could you provide a test case for your addition? In this way we could see some practical use case.

File: cocos/ui/UIWidget.cpp [hunk 1]
```diff
@@ -832,10 +833,10 @@ void Widget::onTouchCancelled(Touch *touch, Event *unusedEvent)
 void Widget::pushDownEvent()
 {
     this->retain();
-    if (_touchEventCallback)
-    {
-        _touchEventCallback(this, TouchEventType::BEGAN);
-    }
+	for (auto &iter : _touchEventCallbacks)
```
Comment (zilongshanren) [inline cocos/ui/UIWidget.cpp]: the indentation is wrong here.

File: cocos/ui/UIWidget.cpp [hunk 2]
```diff
@@ -899,9 +900,38 @@ void Widget::addTouchEventListener(Ref *target, SEL_TouchEvent selector)
     _touchEventSelector = selector;
 }
 
-void Widget::addTouchEventListener(const ccWidgetTouchCallback& callback)
+uint32_t Widget::addTouchEventListener(const ccWidgetTouchCallback &callback)
+{
+	uint32_t key = 0;
+	if (!_touchEventCallbacks.empty())
```
Comment (zilongshanren) [inline cocos/ui/UIWidget.cpp]: please use 4 spaces as indentation, it seems that all of your code use 8 spaces as indentation.

**Comment by enhhh** [comment]
I am so sorry about the indentation. Do I need to add a new comment ? 

I had written a  blog about the use of my addition(in Chinese).
http://enhhh.github.io/2015/05/12/about-newbie-guide/

**Pull request closed without merging**